src/main/java/ca/ciccc/BankAccount.java

```diff
@@ -93,37 +96,87 @@ public void withdraw(double amount) {
      *
      * Then the account would be deducted $10 + $20 + $30 + $40 for the four transactions, leaving
      * a final balance of $0.00. The method would return false.
-     *
+     * */
+
+    /**
      * @param fee
      * @return true if there's enough balance, otherwise false
      */
+
+
+
     public boolean transactionFee(double fee) {
         // TODO 2: Your code goes here.
 
+        // 1. get the total fee
+
+        double feeAmount = 0;
+        for (int i = 0; i <= transactionCount; i++) {
+            feeAmount = i * fee;
+            feeAmount += feeAmount;
+        }
+
+        // 2. check if the account has enough money to pay the total fee
+        if (balance >= feeAmount) {
+            return true;
+        }
+
+        // 3. if you have, return true. Otherwise return false.
         return false;
     }
 
+
     /**
      * The transfer method moves money from `this` bank account to another account. The method
      * accepts two parameters: the amount of money to transfer and the other ca.ciccc.BankAccount to accept
      * the money.
      * There is a $5.00 fee for transferring money, so this much must be deducted from the current
      * account's balance before any transfer.
-     *
+     * <p>
      * If `this` account object does not have enough money to make full transfer, then transfer
      * whatever money is left after the $5.00 fee is deducted. If this account has under $5.00 or amount is 0 or less,
      * no transfer should occur and neither account's state should be modified.
-     *
+     * <p>
      * If any amount of money is transferred, return {@code true}. Otherwise {@code false}.
      *
      * @param amount
      * @param other
      * @return true if transferred any amount of money, otherwise false.
      */
+
     public boolean transfer(double amount, BankAccount other) {
         // TODO 3: Your code goes here.
 
-        return false;
+        // If this account has under $5.00, no transfer should occur and neither account's state should be modified.
+        if (balance >= 5) {
+
+            // There is a $5.00 fee for transferring money, so this much must be deducted from the current account's balance before any transfer.
+            balance -= 5;
+
+            //If any amount of money is transferred, return {@code true}.
+            if (balance >= amount) {
+
+                // transfer: moves money from `this` bank account to another account.
+                balance -= amount;
+                other.deposit(amount);
```

Comment:
[FYI] I think deposit and transferring balance are different. Because the `other` will take a transaction fee as well as 5.00 transferring fee even though the hero doesn't take it. You can just change the `other`'s balance instead.

```suggestion
                other.balance += amount;
```